sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -110,15 +111,33 @@ def _prepare_structure_download(self, files_base_path, files_folder_path, file_n
             else:
                 self.aws_paths.append(f"{file_name}")
 
-    def download(self, bucket_name, files_base_path, files_folder_path, file_names, bucket_subfolder):
+    def download(self, bucket_name, files_base_path, files_folder_path,
+                 file_names, bucket_subfolder, preserve_downloaded_folder_structure=False):
         self._prepare_structure_download(files_base_path, files_folder_path, file_names, bucket_subfolder)
         download_succeded = False
         directory = os.path.join(files_base_path, files_folder_path)
         if not os.path.exists(directory):
             os.makedirs(directory)
-        for file_full_path, aws_path in zip(self.file_full_paths, self.aws_paths):
+        for file_full_path, aws_path in tqdm(zip(self.file_full_paths, self.aws_paths),
+                                             total=len(self.file_full_paths)):
+            if preserve_downloaded_folder_structure:
+                if not os.path.exists(os.path.join(directory, aws_path)):
+                    os.makedirs(os.path.join(directory, aws_path))
+                file_full_path = os.path.join(files_base_path, files_folder_path, aws_path)
+                if os.path.isdir(file_full_path):
+                    os.rmdir(file_full_path)
             try:
-                self._client.download_file(bucket_name, aws_path, file_full_path)
+                object_size = self._client.head_object(Bucket=bucket_name, Key=aws_path)['ContentLength']
+                with tqdm(total=object_size,
+                          unit="B", unit_scale=True,
+                          desc=file_full_path.replace(directory, ''),
+                          leave=False) as pbar:
+                    self._client.download_file(
+                        Bucket=bucket_name,
+                        Key=aws_path,
+                        Filename=file_full_path,
+                        Callback=lambda bytes_tx: pbar.update(bytes_tx),  # pylint: disable=W0640,W0108
```

Comment:
Why use a lamba function over `Callback=pbar.update`?

Reply:
also interested in this question. appart tfrom the facts that lambdas a just neat